sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java

```diff
@@ -194,14 +224,12 @@ public void addActive(W window) {
   @Override
   public void remove(W window) {
     Set<W> stateAddressWindows = activeWindowToStateAddressWindows.get(window);
-    if (stateAddressWindows == null) {
-      // Window is no longer active.
-      return;
-    }
-    for (W stateAddressWindow : stateAddressWindows) {
-      windowToActiveWindow.remove(stateAddressWindow);
+    if (stateAddressWindows != null) {
+      for (W stateAddressWindow : stateAddressWindows) {
+        windowToActiveWindow.remove(stateAddressWindow);
+      }
+      activeWindowToStateAddressWindows.remove(window);
```

Comment:
Could you comment on why this is ordered this way? It seems like it just reordered things. Also, I think you can do:

```
Set<W> stateAddressWindows = activeWindowToStateAddressWindows.remove(window);
if (stateAddressWindows != null) {
  ...
}
```

And remove the separate calls to `get` and `remove`